File: BackendAPI/Application/DTOs/Output/OrderKoperOutputDto.cs
using Application.Common.Models;
using Domain.Entities;
using Domain.Enums;

namespace Application.DTOs.Output;

public class OrderKoperOutputDto
{
    public required Guid Id { get; set; }
    public required DateTimeOffset CreatedAt { get; set; }
    public required OrderStatus Status { get; set; }
    public required DateTimeOffset? ClosedAt { get; set; }
    public required int Quantity { get; set; }
    public required decimal TotalPrice { get; set; }
    public required List<OrderProductOutputDto> Products { get; set; }
    public required KwekerInfo KwekerInfo { get; set; }
    public required KoperInfo KoperInfo { get; set; }
}

File: FrontendWeb/src/declarations/dtos/output/OrderKoperOutputDto.ts
// OrderKwekerOutputDto.ts
import {OrderStatus} from '../../enums/OrderStatus';
import {KoperInfoOutputDto} from './KoperInfoOutputDto';
import {KwekerInfoOutputDto} from './KwekerInfoOutputDto';
import {OrderProductOutputDto} from './OrderProductOutputDto';

export interface OrderKoperOutputDto {
	id: string;
	createdAt: string;
	status: OrderStatus;
	closedAt: string | null;
	quantity: number;
	totalPrice: number;
	products: OrderProductOutputDto[];
	koperInfo: KoperInfoOutputDto;
	kwekerInfo: KwekerInfoOutputDto;
}
